src/v/config/configuration.cc

```diff
@@ -966,7 +966,7 @@ configuration::configuration()
       "kafka_nodelete_topics",
       "Prevents the topics in the list from being deleted via the kafka api",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
-      {"__audit", "__consumer_offsets", "__redpanda_e2e_probe", "_schemas"})
+      {"__audit", "__consumer_offsets", "_schemas"})
```

Comment:
@graphcareful any implications on metering?

Reply:
Nope, metering has its own hardcoded list of topics:
```
static const auto usage_excluded_topics = std::to_array(
  {model::topic("_schemas"),
   model::topic("__audit"),
   model::topic("__redpanda_e2e_probe")});
```

Reply:
So is this PR good to go, @dotnwat and @graphcareful ?